src/contexts/GameContext.tsx

```diff
@@ -18,96 +18,166 @@ type Action =
-    )
-  })
-
-  return hit
-}
+  export const handleHit = (option: Option, optionList: Option[]) => {
+    for (const item of optionList) {
+      if (item === option) {
+        continue;
+      }
+  
+      const dx = item.x - option.x;
+      const dy = item.y - option.y;
+      const d = Math.sqrt(dx * dx + dy * dy);
+      
+      if (d < 20) {
+        const relativeVelX = item.velh - option.velh;
+        const relativeVelY = item.velv - option.velv;
+  
+        const normalX = dx / d;
+        const normalY = dy / d;
+  
+        const relativeVelocity = relativeVelX * normalX + relativeVelY * normalY;
+  
+        if (relativeVelocity < 0) {
+          const elasticity = 0.9;
+  
+          const impulse = (2 * relativeVelocity) / (1 / item.mass + 1 / option.mass);
+  
+          item.velh -= impulse * normalX / item.mass * elasticity;
+          item.velv -= impulse * normalY / item.mass * elasticity;
+          option.velh += impulse * normalX / option.mass * elasticity;
+          option.velv += impulse * normalY / option.mass * elasticity;
+        }
+        
+        return true
+      }
+    }
+    return false;
+  };
+  
 
 const reducer = (state: OptionsState, action: Action) => {
   switch (action.type) {
-  case ActionType.CREATE_OPTIONS: {
-    const options = ['scissors', 'rocks', 'papers']
-    const optionsState: OptionsState = {
-      scissors: [],
-      rocks: [],
-      papers: []
-    }
+    case ActionType.CREATE_OPTIONS: {
+      const options = ['scissors', 'rocks', 'papers']
+      const optionsState: OptionsState = {
+        scissors: [],
+        rocks: [],
+        papers: []
+      }
 
-    const { left, top, right, bottom } = action
-    if (!left || !top || !right || !bottom) return state
-
-    options.forEach(option => {
-      for (let i = 0; i < 10; i++) {
-        const x = Math.floor(Math.random() * (right - left)) + left
-        const y = Math.floor(Math.random() * (bottom - top)) + top
-        const id = `${option}-${i}`
-        const value = option === 'scissors' ? '✂️' : option === 'rocks' ? '🪨' : '📄'
-        const optionObj = { value, x, y, id }
-
-        if (option === 'scissors') {
-          optionsState.scissors.push(optionObj)
-        } else if (option === 'rocks') {
-          optionsState.rocks.push(optionObj)
-        } else {
-          optionsState.papers.push(optionObj)
+      const { left, top, right, bottom } = action
+      if (!left || !top || !right || !bottom) return state
+
+      options.forEach(option => {
+        for (let i = 0; i < 10; i++) {
+          const x = Math.floor(Math.random() * (right - left)) + left
+          const y = Math.floor(Math.random() * (bottom - top)) + top
+          const id = `${option}-${i}`
+          const acelv = 0
+          const acelh = 0
+          const velh = 0
+          const velv = 0
+          const mass = Math.random() * 10
+          const value = option === 'scissors' ? '✂️' : option === 'rocks' ? '🪨' : '📄'
```

Comment:
Just a funny change.

In JavaScript we can use `.size` to get the number of bytes of data contained within a `Blob`. My suggestion is:
```suggestion
          const value = option === 'scissors' ? '✂️' : option === 'rocks' ? '🪨' : '📄'
          const mass = new Blob([value]).size
```

It's better than using `Math.random` for that, I think.

Reply:
I liked the idea! Idk if it's a problem, but the scissors will have more mass than others.
value, mass
✂️ 6
🪨 4
📄 4

it's interesting to see the scissors having more size, because have more details

Reply:
> I liked the idea! Idk if it's a problem, but the scissors will have more mass than others. value, mass ✂️ 6 🪨 4 📄 4
> 
> it's interesting to see the scissors having more size, because have more details

oh yeah I saw that! I think it's okay to have the scissors with a different size, no problem.